src/features/onboarding/OnboardingSheet.tsx

```diff
@@ -0,0 +1,362 @@
+    const snapPoints = useMemo(() => {
+      if (Platform.OS === 'ios') {
+        return [wh - top - spacing[20]]
+      }
+
+      return [wh - top - spacing[20] - spacing[2]]
+    }, [top, spacing])
+
+    const onBack = useCallback(() => {
+      if (carouselRef.current?.currentIndex === 0) {
+        bottomSheetRef.current?.close()
+        return
+      }
+
+      carouselRef.current?.snapToPrev()
+    }, [carouselRef, bottomSheetRef])
+
+    const slides = useMemo(() => {
+      const pages: CarouselItem[] = []
+
+      if (hookValues.flowType === 'sub-wallet') {
+        pages.push({
+          Screen: AccountAssignScreen,
+        })
+      }
+
+      if (hookValues.flowType === 'command-line') {
+        pages.push(
+          {
+            Screen: CLIAccountImportStartScreen,
+          },
+          {
+            Screen: CLIQrScanner,
+          },
+          {
+            Screen: CLIPasswordScreen,
+          },
+          {
+            Screen: AccountAssignScreen,
+          },
+        )
+      }
+
+      if (hookValues.flowType === 'secret-phrase') {
+        pages.push(
+          ...[
+            {
+              Screen: AccountImportScreen,
+            },
+            {
+              Screen: ImportSubAccountsScreen,
+            },
+            {
+              Screen: AccountAssignScreen,
+            },
+          ],
+        )
+      }
+
+      if (hookValues.flowType === 'private-key') {
+        pages.push(
+          ...[
+            {
+              Screen: ImportPrivateKey,
+            },
+            {
+              Screen: ImportSubAccountsScreen,
+            },
+            {
+              Screen: AccountAssignScreen,
+            },
+          ],
+        )
+      }
+
+      if (hookValues.flowType === 'create-account') {
+        pages.push(
+          {
+            Screen: AccountCreatePassphraseScreen,
+          },
+          {
+            Screen: AccountEnterPassphraseScreen,
+          },
+          {
+            Screen: AccountAssignScreen,
+          },
+        )
+      }
+
+      if (hookValues.flowType === 'ledger') {
+        if (Platform.OS === 'android') {
+          pages.push(
+            ...[
+              {
+                Screen: DeviceChooseType,
+              },
+            ],
+          )
+        }
+
+        pages.push(
+          ...[
+            {
+              Screen: DeviceScan,
+            },
+            {
+              Screen: DeviceShow,
+            },
+            {
+              Screen: PairSuccess,
+            },
+          ],
+        )
+      }
+
+      if (hookValues.flowType === 'keystone') {
+        pages.push(
+          ...[
+            {
+              Screen: ScanQrCodeScreen,
+            },
+            {
+              Screen: SelectKeystoneAccountsScreen,
+            },
+            {
+              Screen: KeystoneAccountAssignScreen,
+            },
+          ],
+        )
+      }
+
+      return pages
+    }, [hookValues?.flowType])
```

Comment:
nit: Turn the options into a object. Unless it can match multiple then ignore

```js
const FLOW_TYPE_SCREENS = {
    'sub-wallet': [
        { Screen: AccountAssignScreen },
    ],
    ...etc
}

const slides = useMemo(() => {
  const flowType = hookValues.flowType as keyof typeof FLOW_TYPE_SCREENS;
  return FLOW_TYPE_SCREENS[flowType] || [];
}, [hookValues.flowType]);
```